add job_spm_single_results_display.m

askrzatek opened the pull request:
Un job spam pour l'affichage et l'enregistrement automatique des images du premier niveau d'analyse IRMf de chaque sujet.

askrzatek committed:
Create job_spm_single_results_display.m

spm job to display and save screenshots of individual first level activations in different ROIs in the output directory

askrzatek committed:
moved to /matvol/SPM/

moved to /matvol/SPM/

benoitberanger commented on SPM/job_spm_single_results_display.m:
```diff
@@ -0,0 +1,85 @@
+    addpath(wd)
+    addpath(output_dir)
+    spm('defaults','FMRI');
+        matlabbatch{1}.spm.stats.results.spmmat = fspm;
```
Need a loop for several fspm

askrzatek replied:
Done

benoitberanger commented on SPM/job_spm_single_results_display.m:
```diff
@@ -0,0 +1,85 @@
+        
+        matlabbatch{1}.spm.stats.results.conspec.contrasts = Ci; % learn how to choose multiple contrasts or do a Contrast loop
+        matlabbatch{1}.spm.stats.results.conspec.threshdesc = 'none';
+        matlabbatch{1}.spm.stats.results.conspec.thresh = 0.001;
```
need a par structure like all other matvol functions

askrzatek replied:
Done

benoitberanger commented on SPM/job_spm_single_results_display.m:
```diff
@@ -0,0 +1,85 @@
+    %% Load section == 3DT1 normalized
+
+
+ %       try
```
remove unnecessary / testing comments

benoitberanger commented on SPM/job_spm_single_results_display.m:
```diff
@@ -0,0 +1,85 @@
+%   WD : working directory path where to find this script
+%
+%%
+    addpath(wd)
```
avoid using addpath

benoitberanger commented on SPM/job_spm_single_results_display.m:
```diff
@@ -0,0 +1,85 @@
+
+                graphName = char(addsuffixtofilenames(addsuffixtofilenames(Coordlist.names(iROI),'_'),spmfile.xCon(Ci).name));
+                
+                global st
```
cite where does this code comes from, which spm function ?

benoitberanger commented on SPM/job_spm_single_results_display.m:
```diff
@@ -0,0 +1,85 @@
+function job_spm_single_results_display(fspm, Ci, t1, Coordlist, output_dir, spmfile, wd)
+%%  INPUTS
+%   FSPM : cell variable defining the path to the SPM.mat file
+%   CI : integer variable with the number of the contrast to display
```
maybe it's possible to give the "name" of the contrast, I must be possible to compare with the content of SPM.mat file

benoitberanger commented on SPM/job_spm_single_results_display.m:
```diff
@@ -0,0 +1,85 @@
+%   CI : integer variable with the number of the contrast to display
+%   T1 : character variable defining the path to the T1 MRI volume to
+%   display
+%   COORDLIST : structure variable defining ROI's names and coordinates to
```
need an example about how to define this variable

askrzatek committed:
Create job_spm_single_display_test.matlab

askrzatek committed:
Update job_spm_single_results_display.m

askrzatek committed:
Update job_spm_single_display_test.matlab

askrzatek committed:
Update job_spm_single_results_display.m

askrzatek committed:
Delete an unnecessary file

A file being a test of a new version of job_spm_single_display, but not used after all - not needed

askrzatek committed:
Update job_spm_single_results_display.m

Creating the output file architecture - to permit putting the function job_spm_single_results_display in a loop for multiple contrasts rather than doing the contrast loop inside the function

askrzatek committed:
Update job_spm_single_results_display.m

Correction of the "if loop" error for the file architecture in the output directory

askrzatek committed:
Update job_spm_single_results_display.m

correction for the output_dir path  - cannot use get_subdir_reg if directory non-existing, use addsuffixtofilenames instead

askrzatek committed:
Update job_spm_single_results_display.m

Last (I hope) correction to the output architecture

askrzatek committed:
Plenty of updates related to tedana preprocessing.

Merge branch 'master' of git://github.com/romainVala/matvol

askrzatek committed:
Merge branch 'master' of git://github.com/romainVala/matvol

askrzatek committed:
minor changes for regexp  in spm_single_results display

askrzatek committed:
prefix for warped anatomical image change - ^w replacing ^wms

askrzatek committed:
Merge branch 'master' of git://github.com/romainVala/matvol

romainVala reviewed (approved):
un peu tard, mais ... merci !
